src/ruby/puppet-server-lib/puppet/server/certificate.rb

```diff
@@ -15,6 +16,7 @@ def initialize(java_cert)
     end
 
     @java_cert = java_cert
+    @oid_defns = Puppet::Server::Oids.load_custom_oid_file(Puppet[:trusted_oid_mapping_file]) + Puppet::SSL::Oids::PUPPET_OIDS
```

Comment:
It looks like this line would end up getting run every time a `Certificate` object is created, which happens every time a request is handled by the web server.  I think it would be better to have this be a one-time thing.

Maybe the `load_custom_oid_file` call could just be done from the `puppet_config.rb` file like you have below and that could set a class-level variable that has the concatenation of the mappings from the trusted oid mapping file + `Puppet::SSL::Oids::PUPPET_OIDS`.  From the instance methods in this class, then, we could just retrieve the contents of the class-level variable via a self method on `Puppet::Server::Oids`?